src/ECER.Managers.Registry/RegistrantMapper.cs

```diff
@@ -24,25 +25,29 @@ public RegistrantMapper()
     CreateMap<Contract.Registrants.UserProfile, Resources.Accounts.Registrants.UserProfile>()
       .ForMember(d => d.FirstName, opts => opts.MapFrom(s => s.FirstName))
       .ForMember(d => d.LastName, opts => opts.MapFrom(s => s.LastName))
-      .ForMember(d => d.MiddleName, opts => opts.MapFrom(s => s.MiddleName))
+      .ForMember(d => d.MiddleName, opts => opts.MapFrom(s => !string.IsNullOrEmpty(s.RegistrationNumber) ? ECER.Infrastructure.Common.Utility.GetMiddleName(s.FirstName!, s.GivenName!) : s.GivenName))
       .ForMember(d => d.PreferredName, opts => opts.MapFrom(s => s.PreferredName))
       .ForMember(d => d.AlternateContactPhone, opts => opts.MapFrom(s => s.AlternateContactPhone))
       .ForMember(d => d.Email, opts => opts.MapFrom(s => s.Email))
       .ForMember(d => d.DateOfBirth, opts => opts.MapFrom(s => s.DateOfBirth))
       .ForMember(d => d.Phone, opts => opts.MapFrom(s => s.Phone))
+      .ForMember(d => d.IsVerified, opts => opts.MapFrom(s => s.IsVerified))
+      .ForMember(d => d.RegistrationNumber, opts => opts.MapFrom(s => s.RegistrationNumber))
       .ForMember(d => d.ResidentialAddress, opts => opts.MapFrom(s => s.ResidentialAddress))
       .ForMember(d => d.MailingAddress, opts => opts.MapFrom(s => s.MailingAddress))
       .ForMember(d => d.PreviousNames, opts => opts.MapFrom(s => s.PreviousNames))
       .ForMember(d => d.IsRegistrant, opts => opts.MapFrom(s => s.IsRegistrant))
       .ReverseMap()
+      .ForMember(d => d.GivenName, opts => opts.Ignore())
+      .ForMember(d => d.MiddleName, opts => opts.MapFrom(s => s.MiddleName))
```

Comment:
Question 

There's two mappings for middleName here? do they override each other? 

.ForMember(d => d.MiddleName, opts => opts.MapFrom(s => !string.IsNullOrEmpty(s.RegistrationNumber) ? ECER.Infrastructure.Common.Utility.GetMiddleName(s.FirstName!, s.GivenName!) : s.GivenName))

.ForMember(d => d.MiddleName, opts => opts.MapFrom(s => s.MiddleName))

Reply:
we have to get middle name from given name in BCSC account, So, the forward pass uses a conditional mapping, while the reverse pass maps the middle name directly.